code/modules/xenoarcheaology/effect.dm

```diff
@@ -37,23 +63,31 @@
 /datum/artifact_effect/proc/ToggleActivate(var/reveal_toggle = 1)
 	//so that other stuff happens first
 	spawn(0)
```

Comment:
Would be better to change this to a `set waitfor = FALSE` while you're in the area.